File: e2e-tests/playwright/typescript/pages/LoginPage.ts
import type { Locator, Page } from '@playwright/test'
import { FRONTEND_URL } from '../config'

export class LoginPage {
  readonly page: Page
  readonly emailInput: Locator
  readonly passwordInput: Locator
  readonly submitButton: Locator

  constructor(page: Page) {
    this.page = page
    this.emailInput = page.locator('#email-input-field')
    this.passwordInput = page.locator('#password-input-field')
    this.submitButton = page.locator("button[type='submit']")
  }

  async goto() {
    await this.page.goto(`${FRONTEND_URL}/login`)
  }

  async login(email: string, password: string) {
    await this.emailInput.fill(email)
    await this.passwordInput.fill(password)
    const responsePromise = this.page.waitForResponse('**/graphql')
    await this.submitButton.click()
    await responsePromise
  }
}

File: e2e-tests/playwright/java/src/test/java/net/gradido/e2e/pages/LoginPage.java
package net.gradido.e2e.pages;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.microsoft.playwright.Response;

public class LoginPage {
  private final Page page;
  
  private final Locator emailInput;
  private final Locator passwordInput;
  private final Locator submitButton;

  public LoginPage(Page page) {
      this.page = page;
      emailInput = page.locator("input[name='email']");
      passwordInput = page.locator("input[name='password']");
      submitButton = page.locator("button[type='submit']");      
  }

  public void login(String email, String password) {
      emailInput.fill(email);
      passwordInput.fill(password);
      Response response = page.waitForResponse("**/graphql", () -> {
        submitButton.click();
      });   
  }
}